app/src/main/java/com/benefit/services/ProductService.java

```diff
@@ -111,4 +116,16 @@ private boolean isPropertiesMapInProduct(Product product, Map<String, String> pr
         }
         return indicator;
     }
+
+    public RecyclerView.Adapter getUserProductsRecyclerViewAdapter(User user){
+        if (user == null){
+            Log.d(TAG, "Error - user is null. Can't return a UserProductAdapter.");
+            return null;
+        }
+        FirestoreRecyclerOptions<Product> productRecyclerOptions = new FirestoreRecyclerOptions.
+                Builder<Product>().
+                setQuery(productsCollection.whereEqualTo("sellerId", user.getUid()).orderBy("auctionDate", Query.Direction.ASCENDING), Product.class).
```

Comment:
are you sure that you want to order ASCENDING? it will return the oldest products first, and not the newest.